libnfc/drivers/acr122_usb.c

```diff
@@ -638,6 +639,17 @@ acr122_usb_receive(nfc_device *pnd, uint8_t *pbtData, const size_t szDataLen, co
   offset++;
 
   len = abtRxBuf[offset++];
+  status = abtRxBuf[7];
+  error = abtRxBuf[8];
+  if (len == 0 && error == 0xFE) { // ICC_MUTE; XXX check for more errors
+      // Do not check status; my ACR122U seemingly has status=0 in this case,
+      // even though the spec says it should have had bmCommandStatus=1
+      // and bmICCStatus=1.
+      log_put(LOG_GROUP, LOG_CATEGORY, NFC_LOG_PRIORITY_DEBUG, "%s", "Command timed out");
+      pnd->last_error = NFC_ETIMEOUT;
+      return pnd->last_error;
+  }
+
   if (!((len > 1) && (abtRxBuf[10] == 0xd5))) { // In case we didn't get an immediate answer:
```

Comment:
Is `acr122_usb_bulk_read()` initializing the full buffer regardless of the amount of bytes read?